tests/interfaces/messages/handlers/messages-get.spec.ts

```diff
@@ -0,0 +1,241 @@
+    messageCids.push(messageCid);
+
+    reply = await dwn.processMessage(alice.did, recordsDelete.toJSON());
+    expect(reply.status.code).to.equal(202);
+
+    const { protocolsConfigure } = await TestDataGenerator.generateProtocolsConfigure({
+      requester: alice
+    });
+
+    messageCid = await Message.getCid(protocolsConfigure.message);
+    messageCids.push(messageCid);
+
+    reply = await dwn.processMessage(alice.did, protocolsConfigure.toJSON());
+    expect(reply.status.code).to.equal(202);
+
+    const { messagesGet } = await TestDataGenerator.generateMessagesGet({
+      requester: alice,
+      messageCids
+    });
+
+    const messagesGetReply: MessagesGetReply = await dwn.processMessage(alice.did, messagesGet.toJSON());
+    expect(messagesGetReply.status.code).to.equal(200);
+    expect(messagesGetReply.messages!.length).to.equal(messageCids.length);
+
+    for (const messageReply of messagesGetReply.messages!) {
+      expect(messageReply.messageCid).to.not.be.undefined;
+      expect(messageReply.message).to.not.be.undefined;
+      expect(messageCids).to.include(messageReply.messageCid);
+
+      const cid = await Message.getCid(messageReply.message!);
+      expect(messageReply.messageCid).to.equal(cid);
+    }
+  });
+
+  it('returns message as undefined in reply entry when a messageCid is not found', async () => {
+    const alice = await DidKeyResolver.generate();
+    const { recordsWrite } = await TestDataGenerator.generateRecordsWrite({ requester: alice });
+    const recordsWriteMessageCid = await Message.getCid(recordsWrite.message);
+
+    const { message } = await TestDataGenerator.generateMessagesGet({
+      requester   : alice,
+      messageCids : [recordsWriteMessageCid]
+    });
+
+    // 0 messages expected because the RecordsWrite created above was never stored
+    const reply: MessagesGetReply = await dwn.processMessage(alice.did, message);
+    expect(reply.status.code).to.equal(200);
+    expect(reply.messages!.length).to.equal(1);
+
+    for (const messageReply of reply.messages!) {
+      expect(messageReply.messageCid).to.equal(recordsWriteMessageCid);
+      expect(messageReply.message).to.be.undefined;
+    }
+  });
+
+  it('includes encodedData in reply entry if the data is available and dataSize < threshold', async () => {
+    const alice = await DidKeyResolver.generate();
+
+    const { recordsWrite, dataStream } = await TestDataGenerator.generateRecordsWrite({
+      requester: alice
+    });
+
+    const reply = await dwn.processMessage(alice.did, recordsWrite.toJSON(), dataStream);
+    expect(reply.status.code).to.equal(202);
+
+    const recordsWriteMessageCid = await Message.getCid(recordsWrite.message);
+    const { message } = await TestDataGenerator.generateMessagesGet({
+      requester   : alice,
+      messageCids : [recordsWriteMessageCid]
+    });
+
+    // 1 messages expected because the RecordsWrite created above is not bob's
```

Comment:
wait, where does bob enter into this?

Reply:
oh I see a similar comment in the test below which does have a bob persona. I might just omit this comment